src/test/deploy/functions/validate.spec.ts

```diff
@@ -0,0 +1,119 @@
+import { expect } from "chai";
+import * as fsutils from "../../../fsutils";
+import * as validate from "../../../deploy/functions/validate";
+import * as projectPath from "../../../projectPath";
+import * as FirebaseError from "../../../error";
+import * as sinon from "sinon";
+
+// have to require this because no @types/cjson available
+// tslint:disable-next-line
+const cjson = require("cjson");
+
+describe("validate", () => {
+  describe(".functionsDirectoryExists", () => {
+    it("should not throw error if functions directory is present", () => {
+      const ppath = sinon.stub(projectPath, "resolveProjectPath");
+      ppath.returns("some/path/to/project");
+      const listDir = sinon.stub(fsutils, "dirExistsSync");
+      listDir.returns(true);
+
+      expect(validate.functionsDirectoryExists("cwd", "sourceDirName")).to.not.throw;
```

Comment:
I don't think this is _quite_ right. `.throw` will execute the function passed to it, expecting it not to throw, so I think this needs to be

```typescript
expect(() => {
  validate.functionsDirectoryExists("cwd", "sourceDirName");
}).to.not.throw;
```

Reply:
fixed